### PR TITLE
Sphinx-based Documentation

### LINKED ISSUE #179: Automatically build documentation for metrics
**Is your feature request related to a problem? Please describe.**
After merging #163 we will have working documentation website. Let's add docstrings for functions and classes as a separate page.

**Describe the solution you'd like**
Page / pages with automatically parsed documentation, like its done in PyTorch

### PR DESCRIPTION
**Pull request by denproc**
Closes #179
## Proposed Changes

  - [Documentation](https://piq.readthedocs.io/en/docs-sphinx/index.html) with overview, functional API and classes.
  - Readthedocs autodeployment 
 
Ready for review. @zakajd @snk4tr 

I suggest you to go through the whole documentation to propose any design improvements and docstring changes to be resolved. 

Proposed changes:
- [x] Add link from documentation to github repo
- [x] Add link to the documentation to README
- [x] Add note that we are not Facebook 
- [x] Logo
- [x] Improve the docstrings, which describe functions/classes
- [x] Add citation (#216) 
- [x] Load tables from csv

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 # -- Project information -----------------------------------------------------
 
-project = 'PyTorch Image Quality (PIQ)'
+project = 'PIQ'
```
Comment (snk4tr) [inline docs/source/conf.py:24]: As discussed offline, this change seems to be unnecessary

File: docs/source/index.rst [hunk 2]
```diff
@@ -3,12 +3,17 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-PyTorch Image Quality (PIQ) Documentation
+PIQ Documentation
```
Comment (snk4tr) [inline docs/source/index.rst:6]: As discussed offline, this change seems to be unnecessary

File: docs/source/index.rst [hunk 3]
```diff
@@ -3,12 +3,17 @@
...
 =========================================
 .. image:: https://user-images.githubusercontent.com/15848838/95228385-ed106500-0807-11eb-8584-d3fdbdd47ac7.jpeg
     :alt: piq_logo
 
-PyTorch Image Quality (PIQ) is a collection of measures and metrics for image quality assessment.
+.. note::
+   PIQ (PyTorch Image Quality) is not endorsed by Facebook, Inc.
```
Comment (snk4tr) [inline docs/source/index.rst:13]:
```suggestion
   PyTorch Image Quality (PIQ) is not endorsed by Facebook, Inc.
```

File: docs/source/index.rst [hunk 4]
```diff
@@ -3,12 +3,17 @@
...
+
+   PyTorch, the PyTorch logo and any related marks are trademarks of Facebook, Inc.
+
+PIQ (PyTorch Image Quality) is a collection of measures and metrics for image quality assessment.
```
Comment (snk4tr) [inline docs/source/index.rst]:
```suggestion
PyTorch Image Quality (PIQ) is a collection of measures and metrics for image quality assessment.
```

**Review by snk4tr — CHANGES_REQUESTED** [review]

**Comment by denproc** [comment]
@snk4tr @zakajd 
Final version with resolved checkpoints.
Ready for merge

**Review by snk4tr — APPROVED** [review]
Great job, @denproc!

**Pull request merged**